File: backend/components/metiers/banque_finance/tests/test_utils.js
// test_utils.js – Test des utilitaires Environnement (Node.js/Jest)
const audit = require('../utils/audit');
const i18n = require('../utils/i18n');
const validators = require('../utils/validators');

describe('Utils Environnement', () => {
  it('auditEnvironnement doit retourner un score', () => {
    const result = audit.auditEnvironnement({ statut: 'actif' });
    expect(result).toHaveProperty('score');
    expect(result.score).toBeGreaterThan(0);
  });
  it('i18n doit traduire le texte', () => {
    expect(i18n('Test', 'fr')).toMatch('[FR]');
    expect(i18n('Test', 'en')).toMatch('[EN]');
  });
  it('validators doit valider une entité correcte', () => {
    expect(() => validators({ nom: 'ok', statut: 'actif' })).not.toThrow();
  });
  it('validators doit refuser une entité sans nom', () => {
    expect(() => validators({ statut: 'actif' })).toThrow();
  });
});

File: backend/components/metiers/voyage/tests/test_utils.py
import pytest
from ..utils.audit import auditEnvironnement
from ..utils.i18n import i18n
from ..utils.validators import validate_environnement

def test_audit_environnement():
    result = auditEnvironnement({'statut': 'actif'})
    assert 'score' in result
    assert result['score'] > 0

def test_i18n():
    assert '[FR]' in i18n('Test', 'fr')
    assert '[EN]' in i18n('Test', 'en')

def test_validators_ok():
    validate_environnement({'nom': 'ok', 'statut': 'actif'})

def test_validators_fail():
    with pytest.raises(Exception):
        validate_environnement({'statut': 'actif'})
